README.md
# cse134-hw5

Name: Brendan Devlin
PID: A17171807

URL: [My Website](https://genuine-chimera-b8fc87.netlify.app/)


changelog.md
## Removed
    - Link to the Experiments Page, and removed the page itself.
    - Templates for potential future implementations, a personal message from me and last updated time.
    - Some linebreak elements

## Added
    - Google Analytics to my resume home page, and to personal demos page. Now I am tracking page views, page time, scrolls, outbound clicks, etc. This could be useful to see how much traffic my site is getting, which could affect hosting choices. It could be interesting to see if any potential recruiters (page traffic) are checking me out.
    - More appropriate Output tag where I will output the website's last updated time.
    - JS file for outputting website's last updated time based on the content loading. Also added some CSS to style that area.
    - Another code demo to my personal demos page. It is part of the code needed to do a Huffman Tree encoding in C++.
    - More Books to my favorite book list.
    - A lot of more of my personal philosophy.
    - More skills and goals to the table.

## Changed
    - Fixed some of the responsive CSS for the navigation bar at the top of the page.


methodtest.js
window.addEventListener('DOMContentLoaded', init);
   
function init () {
        //date part
        let date = document.querySelector('#articleDate');
        date.value = new Date();



        //http part
        let buttons = document.getElementsByTagName('button');
        let outputArea = document.getElementById('response');

//POST
        buttons[0].addEventListener('click', () => {            
            outputArea.innerText = '';

            const myForm = document.querySelector('form');
            const formData = new FormData(myForm);

            fetch("https://httpbin.org/post", {method: 'POST', Headers:{'Content-Type': 'application/x-www-form-urlencoded'}, body:  formData})
            .then(response => response.json())
            .then(resultJSON => {
                console.log('POST Success:');
    
                setTimeout(()=>{
                    outputArea.innerText = DOMPurify.sanitize(JSON.stringify(resultJSON, null, 5));
                }, 0);

            })
            .catch(error => {
                console.error('POST Error:', error);
            });


        });


//GET        
        buttons[1].addEventListener('click', () => {
            outputArea.innerText = '';

            const myForm = document.querySelector('form');
            const formData = new FormData(myForm);

            const qString = new URLSearchParams(formData).toString();
    
            let getUrl = `https://httpbin.org/get?${qString}`;

            fetch(getUrl, {method: 'GET', Headers:{'Accept': 'application/json', 'Accept-Encoding':'application/json'} })
            .then(response => response.json()) 
            .then(resultJSON => {
                console.log('GET Success: ');
    
                setTimeout(()=>{
                    outputArea.innerText = DOMPurify.sanitize(JSON.stringify(resultJSON, null, 5));
                }, 0);

            })
            .catch(error => {
                console.error('GET Error: ', error);
            });

            
        });


//PUT        
        buttons[2].addEventListener('click', () => {
            outputArea.innerText = '';

            const myForm = document.querySelector('form');
            const formData = new FormData(myForm);

            fetch('https://httpbin.org/put', {method: 'PUT', Headers:{'Content-Type': 'application/x-www-form-urlencoded'}, body: formData })
            .then(response => response.json()) 
            .then(resultJSON => {
                console.log('PUT Success: ');
    
                setTimeout(()=>{
                    outputArea.innerText = DOMPurify.sanitize(JSON.stringify(resultJSON, null, 5));
                }, 0);

            })
            .catch(error => {
                console.error('PUT Error: ', error);
            });

            
        });


//DELETE    
        buttons[3].addEventListener('click', () => {
            outputArea.innerText = '';

            const myForm = document.querySelector('form');
            const idRec = myForm.querySelector('#idRecord');

            const queryStr = (idRec.value === '') ? `?idRecord=noDelete` : `?idRecord=${idRec.value}`;             //DELETE only using ID number, noDelete would be some flag for server side

            fetch('https://httpbin.org/delete'+queryStr, {method: 'DELETE', Headers:{'Content-Type': 'application/x-www-form-urlencoded'}})
            .then(response => response.json()) 
            .then(resultJSON => {
                console.log('DELETE Success:');
    
                setTimeout(()=>{
                    outputArea.innerText = DOMPurify.sanitize(JSON.stringify(resultJSON, null, 5));
                }, 0);

            })
            .catch(error => {
                console.error('DELETE Error: ', error);
            });

                    
        });

}


styledblog.js
 //initialization function, attempts to get from local storage, if empty, add prepopulated blog posts to local storage and then call loadfromlocalstorage
function init() {

    let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
    let postDate = JSON.parse(localStorage.getItem('Post Dates'));
    let postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

    if(postTitles === null || postDate === null  || postSummary === null || postSummary.length===0 || postDate.length===0 || postTitles.length===0){
        localStorage.setItem('Post Titles', JSON.stringify(['First Post', 'Second Post', 'Trials of the Mind']));
        localStorage.setItem('Post Dates', JSON.stringify(['2023-02-20', '2023-02-21', '2023-02-22']));
        localStorage.setItem('Post Summaries', JSON.stringify(['I made my first blog post today. It went ok.', 'My second blog post went a lot better than my first. Perhaps Practice makes perfect', 'I tried to set up a CRUD application today, I am feeling like... Well you know what...']));

        postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        postDate = JSON.parse(localStorage.getItem('Post Dates'));
        postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

    } 

    loadFromLocalStorage();

}
init();

//helper function to load from local storage and render entries to the screen
function loadFromLocalStorage(){

    let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
    let postDate = JSON.parse(localStorage.getItem('Post Dates'));
    let postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

    let blogPosts = document.getElementsByTagName('output');

    for (let index = 0; index < postTitles.length; index++) {
        blogPosts[0].innerHTML += '<li>' + postTitles[index] +  ': ' + postDate[index] + ' — ' + postSummary[index] + " "  + "<button class='editButton'> Edit </button> " + "<button class='deleteButton'> Delete </button>"  +  ' </li>' + '<hr>';

    }

}  
//using dom to access elements
let blogPosts = document.getElementsByTagName('output');
let deleteButtons = blogPosts[0].getElementsByClassName('deleteButton');
//items' delete buttons
for (let index = 0; index < deleteButtons.length; index++) {
    //clicking delete for each entry brings up user dialog field related to that specific entry by index, now delete buttons are trash icon images
    deleteButtons[index].innerHTML = '<img src="content/images/trashIcon.svg" alt="Trash icon representing the delete button"> '
    deleteButtons[index].addEventListener('click', () =>{
        let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        let postDates = JSON.parse(localStorage.getItem('Post Dates'));
        let postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

        //confirm delete field
        let userDialogField = document.getElementById('userDialogField');
        userDialogField.open = true;
        userDialogField.innerHTML = ` <p> <b> Do you want to delete this entry? </b> </p> <br> 
                                    <button id="delConfirmButton">
                                        Delete
                                    </button>
                                    <button id="cancelButton">
                                        Cancel
                                    </button>`;

        //cancel sub-button for deleting, just closes the field and does nothing to localstorage
        cancelButton.addEventListener('click', ()=>{
            userDialogField.open = false;
            userDialogField.innerHTML = '';

            blogPosts[0].innerHTML = '';
            loadFromLocalStorage();
            //reloading window to rerender changes to localstorage, MPA basically
            window.location.reload();
        });

        //confirm delete sub-button, removes that entry from local storage
        let delConfirmButton = document.getElementById('delConfirmButton');
        delConfirmButton.addEventListener('click', ()=>{
            postTitles.splice(index, 1);
            postDates.splice(index, 1);
            postSummaries.splice(index, 1);

            localStorage.setItem('Post Titles', JSON.stringify(postTitles));
            localStorage.setItem('Post Dates', JSON.stringify(postDates));
            localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));

            //closing field
            userDialogField.open = false;
            userDialogField.innerHTML = '';

            blogPosts[0].innerHTML = '';
            loadFromLocalStorage();
            window.location.reload();
        });
      
    });
    
} 
//items' edit buttons
let editButtons = blogPosts[0].getElementsByClassName('editButton');
for (let index = 0; index < editButtons.length; index++) {
    //clicking edit for each entry brings up user dialog field related to that specific entry by index, now instead of edit button it is a pencil icon.
    editButtons[index].innerHTML = '<img src="content/images/pencilIcon.svg" alt="Pencil icon representing the edit button"> '
    editButtons[index].addEventListener('click', () => {
        let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        let postDates = JSON.parse(localStorage.getItem('Post Dates'));
        let postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

        //getting user field and making it visible
        let userDialogField = document.getElementById('userDialogField');
        userDialogField.open = true;

        //giving entry fields with prepopulated values
        userDialogField.innerHTML = ` <label for="postTitle">Edit Title:</label>
                    <input type="text" id="postTitle" name="postTitle" value="${postTitles[index]}" required>
    
                    <label for="Date">Edit Date:</label>
                    <input type="date" id="Date" name="Date" value=${postDates[index]} required>
                    <br>
                    <label for="summary">Edit Summary:</label> <br>
                    <textarea id="summary" name="summary" required> ${postSummaries[index]}  </textarea>
                    <br>
                    <button id="saveButton">
                        Save
                    </button>
                    <button id="cancelButton">
                        Cancel
                    </button>`;

        //cancel sub-button in the user dialog field for edit
        cancelButton.addEventListener('click', ()=>{
            userDialogField.open = false;
            userDialogField.innerHTML = '';

            blogPosts[0].innerHTML = '';
            loadFromLocalStorage();
            window.location.reload();
        });

        //save sub-button in the user dialog field, saving edits for that user given entry to local storage in the same index
        saveButton.addEventListener('click', ()=>{
            let postTitleEle = document.getElementById('postTitle');
            let dateEle = document.getElementById('Date');
            let summaryEle = document.getElementById('summary');
    
            const postTitles = JSON.parse(localStorage.getItem('Post Titles'));
            const postDates = JSON.parse(localStorage.getItem('Post Dates'));
            const postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

            //editing and sanitizing user inputs
            postTitles[index] = DOMPurify.sanitize(postTitleEle.value);
            postDates[index] = DOMPurify.sanitize(dateEle.value);
            postSummaries[index] = DOMPurify.sanitize(summaryEle.value);

            //resaving changes to localStorage
            localStorage.setItem('Post Titles', JSON.stringify(postTitles));
            localStorage.setItem('Post Dates', JSON.stringify(postDates));
            localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));
        
            //closing field
            userDialogField.open = false;
            userDialogField.innerHTML = '';

            blogPosts[0].innerHTML = '';
            loadFromLocalStorage();
            window.location.reload();
        });


    });
    
} 

//app's add button
let addButtons = document.getElementsByClassName('addButton');

addButtons[0].addEventListener('click', () => {
    //clicking add brings up user dialog field
    let userDialogField = document.getElementById('userDialogField');
    userDialogField.open = true;
    userDialogField.innerHTML = ` <label for="postTitle">Post Title:</label>
                <input type="text" id="postTitle" name="postTitle" required>

                <label for="Date">Date:</label>
                <input type="date" id="Date" name="Date" required>
                <br>
                <label for="summary">Summary:</label> 
                <textarea id="summary" name="summary" required> </textarea>
                <br>
                <button id="saveButton">
                    Save
                </button>
                <button id="cancelButton">
                    Cancel
                </button>`;

    let saveButton = document.getElementById('saveButton');
    let cancelButton = document.getElementById('cancelButton');

     //cancel sub-button in the user dialog field, does nothing to local storage
    cancelButton.addEventListener('click', ()=>{
        userDialogField.open = false;
        userDialogField.innerHTML = '';
    });

    //save sub-button in the user dialog field, appending user given entry to local storage 
    saveButton.addEventListener('click', ()=>{
        let postTitleEle = document.getElementById('postTitle');
        let dateEle = document.getElementById('Date');
        let summaryEle = document.getElementById('summary');

        const postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        const postDates = JSON.parse(localStorage.getItem('Post Dates'));
        const postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

        //santitizing user inputs before adding it to local storage
        postTitles.push(DOMPurify.sanitize(postTitleEle.value));
        postDates.push(DOMPurify.sanitize(dateEle.value));
        postSummaries.push(DOMPurify.sanitize(summaryEle.value));

        //setting local storage
        localStorage.setItem('Post Titles', JSON.stringify(postTitles));
        localStorage.setItem('Post Dates', JSON.stringify(postDates));
        localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));

        //closing field
        userDialogField.open = false;
        userDialogField.innerHTML = '';

        document.getElementsByTagName('output')[0].innerHTML = '';
        loadFromLocalStorage();
        window.location.reload();
    });


});




customdialog.js
// getting buttons, dialog fields, and result fields
let buttons = document.getElementsByTagName('button');
let dialogs = document.getElementsByTagName('dialog');
let results = document.getElementsByTagName('result');

//alert button
buttons[0].addEventListener('click', () =>{
    //change background for some contrast, similar to example
    let background = document.getElementsByTagName('html');
    background[0].style = ' background-color:rgb(70, 224, 224);';

    //set result areas to be blank, no style or content
    results[0].innerHTML = "";
    results[0].style = "";
    //give dialog field to user for alert.
    dialogs[0].innerHTML = "Alert Pressed! <br> <button id='alertOkButton'>Ok</button> ";
    dialogs[0].open = true;

    //user hits ok for alert prompt, and we change dialog to show this and change background back to regular, 
    //note I am mimicking the example video where there is only the 'ok' response to the alert dialog field
    let innerOkButton = document.getElementById('alertOkButton');
    innerOkButton.addEventListener('click', () =>{
        //closing dialog for prompt
        dialogs[0].open = false;
        dialogs[0].innerHTML = "";
        background[0].style = 'background-color:rgb(200, 224, 224);';
    });

});

//confirm button
buttons[1].addEventListener('click', () =>{
    //change background for some contrast, similar to example
    let background = document.getElementsByTagName('html');
    background[0].style = ' background-color:rgb(110, 224, 224);';
    //set result areas to be blank, no style or content
    results[0].innerHTML = "";
    results[0].style = "";
    //give dialog field to user to confirm or not
    dialogs[0].innerHTML = "Do you confirm this? <br> <button id='confirmOkButton'>Ok</button> <button id='confirmCancelButton'>Cancel</button> ";
    dialogs[0].open = true;

    //set confirm result field to reflect clicking ok
    let innerOkButton = document.getElementById('confirmOkButton');
        innerOkButton.addEventListener('click', () =>{
        dialogs[0].open = false;
        dialogs[0].innerHTML = "";
        results[0].style = 'border: solid 1px black; font-weight:500; background-color:pink;';
        results[0].innerHTML = confirmResultTemplateStr('true');
        background[0].style = 'background-color:rgb(200, 224, 224);';
    });

    //set confirm result field to reflect clicking cancel
    let innerCancelButton = document.getElementById('confirmCancelButton');
        innerCancelButton.addEventListener('click', () =>{
        //closing dialog for prompt
        dialogs[0].open = false;
        dialogs[0].innerHTML = "";
        results[0].style = 'border: solid 1px black; font-weight:500; background-color:pink;';
        results[0].innerHTML = confirmResultTemplateStr('false');
        background[0].style = 'background-color:rgb(200, 224, 224);';
    });

});

//helper tagged template strings
function confirmResultTemplateStr(result){
    return `Confirm result: ${result}`
}
function promptResultTemplateStr(result){
    return `Prompt result: ${result}`
}

//Prompt button
buttons[2].addEventListener('click', () =>{
    let background = document.getElementsByTagName('html');
    background[0].style = ' background-color:rgb(170, 224, 224);';

    //set result areas to be blank, no style or content
    results[0].innerHTML = "";
    results[0].style = "";
     //give dialog field to user to enter prompt or not
    dialogs[0].innerHTML = "What is your name? <br> <textarea id='promptTextArea'> </textarea> <br> <button id='promptOkButton'>Ok</button> <button id='promptCancelButton'>Cancel</button> ";
    dialogs[0].open = true;

    let innerOkButton = document.getElementById('promptOkButton');

    //user clicks ok on prompt dialog field, grabs text inputted from user and populates result field, catches empty string given case
    innerOkButton.addEventListener('click', () =>{
        let promptTxtArea = document.getElementById('promptTextArea');   
        //sanitizing user input
        let txtFromUser = DOMPurify.sanitize(promptTxtArea.value);
        //if txt from user is blank, no name is given
        txtFromUser = txtFromUser===' ' ? "No name given" : txtFromUser;

        //closing dialog for prompt
        dialogs[0].open = false;
        dialogs[0].innerHTML = "";
        results[0].style = 'border: solid 1px black; font-weight:500; background-color:pink';
        //setting results values to txt from user.
        results[0].innerHTML = promptResultTemplateStr(txtFromUser);
        background[0].style = 'background-color:rgb(200, 224, 224);';
    });

    //user click cancel on prompt dialog field, populates results with that information
    let innerCancelButton = document.getElementById('promptCancelButton');
    innerCancelButton.addEventListener('click', () =>{
        //close dialog field
        dialogs[0].open = false;
        dialogs[0].innerHTML = "";
        results[0].style = 'border: solid 1px black; font-weight:500; background-color:pink';
        
        //populate result field to reflect user hit cancel
        results[0].innerHTML = promptResultTemplateStr('No name given, User hit cancel');
        background[0].style = 'background-color:rgb(200, 224, 224);';
    });

});


nativedialogs.js
  //get buttons
        let buttons = document.getElementsByTagName('button');
  //built in alert part
        buttons[0].addEventListener('click', () => {
            outputs[0].innerHTML = '';

            setTimeout(() => {
                alert('Built-in Alert')
            }, 10);
        
        } );

        let outputs = document.getElementsByTagName('output');

        let userValue = null;

       //use of tagged template strings 
        function taggedTemplateStr (str){       
            if(str === null || str.length < 1){
                return `Prompt Result : User did not enter anything`;
            }
            
            return `Prompt Result : ${str}`;
        }
  //confirm button with response code    
        buttons[1].addEventListener('click', () => {
            outputs[0].innerHTML = '';

            setTimeout(() => {
                let confirmation = confirm('Do you confirm this?');
                let strToReturn = `Confirm Result: ${confirmation}`;
                outputs[0].innerHTML = strToReturn;
            }, 10);
           
        } );
  //regular prompt button with response code    
        buttons[2].addEventListener('click', () => {        //Regular Prompt
            outputs[0].innerHTML = '';

            setTimeout(() => {
                userValue = prompt('Please enter your name!')
                outputs[0].innerHTML = taggedTemplateStr(userValue);
            }, 10);

        });
  //safer prompt button with response code    
        buttons[3].addEventListener('click', () => {        //Safer Prompt
            outputs[0].innerHTML = '';

            setTimeout(() => {
                userValue = prompt('Please enter your name!');
                userValue = DOMPurify.sanitize(userValue);
                outputs[0].innerHTML = taggedTemplateStr(userValue);
            }, 10);
            
        });

                                //string for testing: <b onmouseover="alert('Got \'Em')"> Roll them </b>

blog.js
    //initialization function, attempts to get from local storage, if empty, add prepopulated blog posts to local storage and then call loadfromlocalstorage
    function init() {

        let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        let postDate = JSON.parse(localStorage.getItem('Post Dates'));
        let postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

        if(postTitles === null || postDate === null  || postSummary === null || postSummary.length===0 || postDate.length===0 || postTitles.length===0){
            localStorage.setItem('Post Titles', JSON.stringify(['First Post', 'Second Post', 'Trials of the Mind']));
            localStorage.setItem('Post Dates', JSON.stringify(['2023-02-20', '2023-02-21', '2023-02-22']));
            localStorage.setItem('Post Summaries', JSON.stringify(['I made my first blog post today. It went ok.', 'My second blog post went a lot better than my first. Perhaps Practice makes perfect', 'I tried to set up a CRUD application today, I am feeling like... Well you know what...']));

            postTitles = JSON.parse(localStorage.getItem('Post Titles'));
            postDate = JSON.parse(localStorage.getItem('Post Dates'));
            postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

        } 

        loadFromLocalStorage();
    
    }
    init();

    //helper function to load from local storage and render entries to the screen
    function loadFromLocalStorage(){

        let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        let postDate = JSON.parse(localStorage.getItem('Post Dates'));
        let postSummary = JSON.parse(localStorage.getItem('Post Summaries'));

        let blogPosts = document.getElementsByTagName('output');

        for (let index = 0; index < postTitles.length; index++) {
            blogPosts[0].innerHTML += '<li>' + postTitles[index] +  ': ' + postDate[index] + ' — ' + postSummary[index] + " |"  + "<button class='editButton'> Edit </button> " + "<button class='deleteButton'> Delete </button>"  +  ' </li>';

        }
        

    }  

//using dom to access elements
    let blogPosts = document.getElementsByTagName('output');
    let deleteButtons = blogPosts[0].getElementsByClassName('deleteButton');
//items' delete buttons
    for (let index = 0; index < deleteButtons.length; index++) {
        //clicking delete for each entry brings up user dialog field related to that specific entry by index
        deleteButtons[index].addEventListener('click', () =>{
            let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
            let postDates = JSON.parse(localStorage.getItem('Post Dates'));
            let postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

            //confirm delete field
            let userDialogField = document.getElementById('userDialogField');
            userDialogField.open = true;
            userDialogField.innerHTML = ` <p> <b> Do you want to delete this entry? </b> </p> <br> 
                                        <button id="delConfirmButton">
                                            Delete
                                        </button>
                                        <button id="cancelButton">
                                            Cancel
                                        </button>`;

            //cancel sub-button for deleting, just closes the field and does nothing to localstorage
            cancelButton.addEventListener('click', ()=>{
                userDialogField.open = false;
                userDialogField.innerHTML = '';

                blogPosts[0].innerHTML = '';
                loadFromLocalStorage();
                window.location.reload();
            });

            //confirm delete sub-button, removes that entry from local storage
            let delConfirmButton = document.getElementById('delConfirmButton');
            delConfirmButton.addEventListener('click', ()=>{
                postTitles.splice(index, 1);
                postDates.splice(index, 1);
                postSummaries.splice(index, 1);

                localStorage.setItem('Post Titles', JSON.stringify(postTitles));
                localStorage.setItem('Post Dates', JSON.stringify(postDates));
                localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));

                //closing field
                userDialogField.open = false;
                userDialogField.innerHTML = '';

                blogPosts[0].innerHTML = '';
                loadFromLocalStorage();
                //reloading window to rerender changes to localstorage, MPA basically
                window.location.reload();
            });

          
        });
        
    } 
//items' edit buttons
    let editButtons = blogPosts[0].getElementsByClassName('editButton');
    //clicking edit for each entry brings up user dialog field related to that specific entry by index
    for (let index = 0; index < editButtons.length; index++) {
        editButtons[index].addEventListener('click', () => {
            //parsing from local storage
            let postTitles = JSON.parse(localStorage.getItem('Post Titles'));
            let postDates = JSON.parse(localStorage.getItem('Post Dates'));
            let postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));

            //getting user field and making it visible
            let userDialogField = document.getElementById('userDialogField');
            userDialogField.open = true;

            //giving entry fields with prepopulated values
            userDialogField.innerHTML = ` <label for="postTitle">Edit Title:</label>
                        <input type="text" id="postTitle" name="postTitle" value="${postTitles[index]}" required>
        
                        <label for="Date">Edit Date:</label>
                        <input type="date" id="Date" name="Date" value=${postDates[index]} required>
                        <br>
                        <label for="summary">Edit Summary:</label> <br>
                        <textarea id="summary" name="summary" required> ${postSummaries[index]} </textarea>
                        <br>
                        <button id="saveButton">
                            Save
                        </button>
                        <button id="cancelButton">
                            Cancel
                        </button>`;


            //cancel sub-button in the user dialog field for edit
            cancelButton.addEventListener('click', ()=>{
                userDialogField.open = false;
                userDialogField.innerHTML = '';

                blogPosts[0].innerHTML = '';
                loadFromLocalStorage();
                window.location.reload();
            });

            //save sub-button in the user dialog field, saving edits for that user given entry to local storage in the same index
            saveButton.addEventListener('click', ()=>{
                let postTitleEle = document.getElementById('postTitle');
                let dateEle = document.getElementById('Date');
                let summaryEle = document.getElementById('summary');
        
                const postTitles = JSON.parse(localStorage.getItem('Post Titles'));
                const postDates = JSON.parse(localStorage.getItem('Post Dates'));
                const postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));
    
                //editing and sanitizing user inputs
                postTitles[index] = DOMPurify.sanitize(postTitleEle.value);
                postDates[index] = DOMPurify.sanitize(dateEle.value);
                postSummaries[index] = DOMPurify.sanitize(summaryEle.value);

                localStorage.setItem('Post Titles', JSON.stringify(postTitles));
                localStorage.setItem('Post Dates', JSON.stringify(postDates));
                localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));
            
                //closing field
                userDialogField.open = false;
                userDialogField.innerHTML = '';

                blogPosts[0].innerHTML = '';
                loadFromLocalStorage();
                window.location.reload();
            });


        });
        
    } 

//app's add button
  let addButtons = document.getElementsByClassName('addButton');
    //clicking add brings up user dialog field
  addButtons[0].addEventListener('click', () => {
    let userDialogField = document.getElementById('userDialogField');
    userDialogField.open = true;
    //giving default fields for adding to blog posts
    userDialogField.innerHTML = ` <label for="postTitle">Post Title:</label>
                <input type="text" id="postTitle" name="postTitle" required>

                <label for="Date">Date:</label>
                <input type="date" id="Date" name="Date" required>
                <br>
                <label for="summary">Summary:</label> 
                <textarea id="summary" name="summary" required> </textarea>
                <br>
                <button id="saveButton">
                    Save
                </button>
                <button id="cancelButton">
                    Cancel
                </button>`;

    let saveButton = document.getElementById('saveButton');
    let cancelButton = document.getElementById('cancelButton');

     //cancel sub-button in the user dialog field, field closes and local storage is unchanged.
    cancelButton.addEventListener('click', ()=>{
        userDialogField.open = false;
        userDialogField.innerHTML = '';
    });

    //save sub-button in the user dialog field, appending user given entry to local storage 
    saveButton.addEventListener('click', ()=>{
        let postTitleEle = document.getElementById('postTitle');
        let dateEle = document.getElementById('Date');
        let summaryEle = document.getElementById('summary');

        const postTitles = JSON.parse(localStorage.getItem('Post Titles'));
        const postDates = JSON.parse(localStorage.getItem('Post Dates'));
        const postSummaries = JSON.parse(localStorage.getItem('Post Summaries'));


        postTitles.push(DOMPurify.sanitize(postTitleEle.value));
        postDates.push(DOMPurify.sanitize(dateEle.value));
        postSummaries.push(DOMPurify.sanitize(summaryEle.value));

        localStorage.setItem('Post Titles', JSON.stringify(postTitles));
        localStorage.setItem('Post Dates', JSON.stringify(postDates));
        localStorage.setItem('Post Summaries', JSON.stringify(postSummaries));
    
        //closing field
        userDialogField.open = false;
        userDialogField.innerHTML = '';

        document.getElementsByTagName('output')[0].innerHTML = '';
        loadFromLocalStorage();
        window.location.reload();
    });



  });


